### DISCUSSION TITLE
the mass and moment of inertia information for the whole system

### DISCUSSION THREAD

**Discussion by YangLeiTH** [OpenFAST/openfast#1861] [General]
Hi @jjonkman,

I am writing to seek your assistance regarding the use of Hydrostar software for calculating the natural period of OC4, and comparing it with the natural period obtained from Openfast linearization. However, I am currently facing some confusion regarding how to obtain the mass and moment of inertia information for the whole system.

I was wondering if there are any documentation or resources available that can help me understand how to obtain the mass information for the entire system using Openfast. Additionally, if there are any other tools or methods that you would recommend for obtaining these values, I would greatly appreciate your suggestions.

Thank you for taking the time to read my email, and I look forward to hearing back from you soon.

Best regards,

**Comment by YangLeiTH** [comment]
Dear @jjonkman,

I am writing to inquire about the discrepancy I have noticed in the official benchmark case for 5MW_OC4Semi_Linear. Specifically, when performing linearization on the 5MW_OC4Semi_Linear.1.lin file located in the 5MW_OC4Semi_Linear folder, I found that the natural frequencies in the roll and pitch are 0.047Hz. However, in the "Definition of the Semisubmersible Floating System for Phase II of OC4," it is stated that the natural frequency in the platform pitch should be 0.037Hz.

Could you please explain why there is a difference between these two values? I would greatly appreciate any clarification or additional information you can provide.

Best regards,

**Comment by YangLeiTH** [comment]
I have noticed that when I perform linearization on the 5MW_OC4Semi_Linear.1.lin file downloaded from the official website, the natural periods of roll and pitch are 0.039Hz. However, when I re-run the 5MW_OC4Semi_Linear.fst file and perform linearization on the resulting 5MW_OC4Semi_Linear.1.lin file, the natural period is indeed 0.047Hz. I am wondering why this difference occurs.

**Comment by ebranlard** [comment]
Hi @YangLeiTH 
To extract the rigid body inertia of the turbine or the RNA, you can use the following [python script](https://github.com/ebranlard/welib/blob/main/welib/fast/examples/Example_ExtractInertia.py), 
which uses functions defined in [the file `extract.py`](https://github.com/ebranlard/welib/blob/main/welib/fast/extract.py). 
To use it, you'll have to you to install the library [welib](https://github.com/ebranlard/welib). If you have error running the script, you can post an issue on the `welib` repository. 

You can look in the file `extract.py` to see what is being done by the python script. The idea is to use linearization of the structure, while the platform degrees of freedom are on and look at the input output relationships (in the lin file) between platform forces and acceleration. The Python script automatize this procedure, but you can try to do it yourself as well. 

I cannot comment on the differences in frequencies you see, they could come from small differences in the input file (platform inertia, tower files, mooring, hydrodynamics) or differences in the OpenFAST implementation itself.

I hope that helps,

Emmanuel

**Comment by jjonkman** [comment]
Dear @YangLeiTH,

Regarding how to extract the rigid-body mass matrix from OpenFAST, @ebranlard scripts look quite powerful.  The following question and answer on our forum are likely of interest to you as well: https://forums.nrel.gov/t/openfast-2nd-order-linearization/2249.

Regarding linearization of the OC4-DeepCwind semi, you report different frequencies depending on the source.  When you say, "5MW_OC4Semi_Linear.1.lin file downloaded from the official website", are you referring to the following file: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC4Semi_Linear?  Can you clarify what is different between your set-up and the set-up used to generate this file?

Best regards,
